extensions/amp-youtube/0.1/test/test-amp-youtube.js

```diff
@@ -291,6 +291,17 @@ describe('amp-youtube', function() {
     });
   });
 
+  it('should remove iframe after unlayoutCallback', () => {
+    return getYt({'data-videoid': 'mGENRKrdoGY'}).then(yt => {
+      const placeholder = yt.querySelector('[placeholder]');
+      const obj = yt.implementation_;
+      obj.unlayoutCallback();
+      expect(yt.querySelector('iframe')).to.be.null;
```

Comment:
let's also make sure `unlistenMessage_` is called. You can do that by installing a `spy` (e.g. `const unlistenSpy = sandbox.spy(obj, 'unlistenMessage_')`) on it before calling  `unlayoutCallback` and then checking that spy has been called via `expect(unlistenSpy).to.have.been.called;`